File: go/442_Find_All_Duplicates_in_an_Array/442_Find_All_Duplicates_in_an_Array.go
func findDuplicates(nums []int) []int {
    if len(nums) == 1 {
        return []int{}
    }
    sort.Ints(nums)
    result := make([]int, 0)
    for i := 0; i < len(nums)-1; i++ {
        if nums[i] == nums[i+1] {
            result = append(result, nums[i])
        }
    }
    return result
}

File: java/442_Find_All_Duplicates_in_an_Array/442_Find_All_Duplicates_in_an_Array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        if (nums.length == 1) {
            return new ArrayList<Integer>();
        }
        Arrays.sort(nums);
        ArrayList<Integer> result = new ArrayList<Integer>();
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] == nums[i+1]) {
                result.add(nums[i]);
            }
        }
        return result;
    }
}
